libavcodec/vaapi_decode.c

```diff
@@ -569,6 +587,22 @@ static int vaapi_decode_make_config(AVCodecContext *avctx,
         frames->width = avctx->coded_width;
         frames->height = avctx->coded_height;
 
+        if (avctx->sub_frame_opts) {
+            AVDictionaryEntry *e = NULL;
+            while ((e = av_dict_get(avctx->sub_frame_opts, "", e, AV_DICT_IGNORE_SUFFIX))) {
+                if (!strcmp(e->key, "width"))
+                    frames->sfc_w = atoi(e->value);
+                else if (!strcmp(e->key, "height"))
+                    frames->sfc_h = atoi(e->value);
+                else if (!strcmp(e->key, "format"))
+                    frames->sfc_format = av_get_pix_fmt(e->value);
+            }
+        }
+
+        frames->enable_sfc =
+            !!((avctx->export_side_data & AV_CODEC_EXPORT_DATA_SUB_FRAME) &&
```

Comment:
What is the purpose of two "!" here

Reply:
This is used to turn none-zero value to 1. I add another condition with &&, so now don't need it anymore. thanks.